libraries/AP_Scheduler/AP_Scheduler.h

```diff
@@ -12,66 +12,60 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-/*
- *  main loop scheduler for APM
- *  Author: Andrew Tridgell, January 2013
- *
- */
 #pragma once
 
+
+#define __FILE FILE
+#include <functional>
+#undef __FILE
+
 #include <AP_Param/AP_Param.h>
 #include <AP_HAL/Util.h>
 #include <AP_Math/AP_Math.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle.h>
 
-#define AP_SCHEDULER_NAME_INITIALIZER(_name) .name = #_name,
-
-/*
-  useful macro for creating scheduler task table
- */
-#define SCHED_TASK_CLASS(classname, classptr, func, _rate_hz, _max_time_micros) { \
-    .function = FUNCTOR_BIND(classptr, &classname::func, void),\
```

Comment:
Did moving to C++ lamdas improve things in any way?  Additional type safety or better compiler warnings?

Personally I'd like to investigate their use some more in our code - but the functor stuff is well-proven and my understanding is that lambdas have problems in various compilers.  Also, it's critical to keep the scheduler table in .text - tridge spent a lot of time making sure that worked across different compilers and targets etc.